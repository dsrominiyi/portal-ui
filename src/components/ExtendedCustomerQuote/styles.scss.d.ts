export const quoteContainer: string;
export const quoteWidget: string;
export const quoteWidgetWide: string;
export const quoteHeader: string;
export const iconContainer: string;
export const userImage: string;
