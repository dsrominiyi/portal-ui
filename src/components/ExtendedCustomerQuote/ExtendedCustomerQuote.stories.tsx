import React from 'react';
import { storiesOf } from '@storybook/react';

import ExtendedCustomerQuote from '.';

import mockData from '../../../__mocks__/mock-data.json';

const stories = storiesOf('ExtendedCustomerQuote', module);

const fullQuote: Quote = mockData.quotes[0];
const withoutPhoto: Quote = {
  ...fullQuote,
  salesExecutive: {
    name: {
      forename: 'James',
      surname: 'Fowler'
    },
    username: 'Jimmy123'
  }
};
const noRate: Quote = {
  ...fullQuote,
  rate: -1
};
const declined: Quote = {
  ...fullQuote,
  type: 'invalid'
};

const render = (quote: Quote) => () => (
  <ExtendedCustomerQuote quote={quote} onClick={() => alert('You clicked!')} />
);

export const storyRenders = {
  fullQuote: render(fullQuote),
  noPhoto: render(withoutPhoto),
  noRate: render(noRate),
  declined: render(declined)
};

stories.add('Full quote', storyRenders.fullQuote);
stories.add('Without photo', storyRenders.noPhoto);
stories.add('Without rate', storyRenders.noRate);
stories.add('Declined quote', storyRenders.declined);
