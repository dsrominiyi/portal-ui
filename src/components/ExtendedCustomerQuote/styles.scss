@import '../../style/vars';

.quote-container {
  display: flex;
  flex-direction: row;
  width: 100%;
  color: $brand-charcoal;
  padding: 24px 32px;
  box-sizing: border-box;
  background-color: $brand-white;
  cursor: pointer;

  &:hover {
    background-color: darken($brand-white, 3%);
  }
}

.quote-widget {
  display: flex;
  flex-basis: 16.66667%;
  flex-direction: column;
  align-items: center;
  padding-right: 12px;
  padding-left: 12px;

  &-wide {
    @extend .quote-widget;

    flex-basis: 25%;
  }
}

.quote-header {
  font-size: 16px;
  margin-top: 8px;
  margin-bottom: 16px;
}

.icon-container {
  svg {
    width: 70px;
    height: 70px;
  }
}

.user-image {
  font-size: 22px;
  height: 70px;
  width: 70px;
}
