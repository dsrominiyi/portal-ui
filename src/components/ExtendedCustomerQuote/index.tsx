import React, { FunctionComponent, MouseEvent } from 'react';
import { utc as moment } from 'moment';
import policyTypeMapper from '../../helpers/policyTypeMapper';
import styles from './styles.scss';
import { policyTypeInfoMap } from '../../helpers/policyTypeInfoMap';
import UserImage from '../UserImage';

interface Props {
  quote: Quote;
  onClick?(event: MouseEvent<HTMLDivElement>): void;
}

export const ExtendedCustomerQuote: FunctionComponent<Props> = ({ quote, onClick }: Props) => {
  const type = policyTypeMapper(quote.type);

  if (!type) {
    return null;
  }

  const { id, startDate, rate, salesExecutive } = quote;
  const { icon, displayName } = policyTypeInfoMap[type];
  const quoteRate = rate > 0 ? `Â£${rate}` : 'No Quote';

  return (
    <div className={styles.quoteContainer} onClick={onClick}>
      <div className={styles.quoteWidget}>
        <div className={styles.quoteHeader}>Reference</div>
        <h4>{id}</h4>
      </div>
      <div className={styles.quoteWidget}>
        <div className={styles.quoteHeader}>{displayName}</div>
        <div className={styles.iconContainer}>{icon}</div>
      </div>
      <div className={styles.quoteWidgetWide}>
        <div className={styles.quoteHeader}>Inception Date</div>
        <h4>{moment(startDate).format('DD/MM/YYYY')}</h4>
      </div>
      <div className={styles.quoteWidget}>
        <div className={styles.quoteHeader}>Sales Executive</div>
        {salesExecutive && (
          <UserImage
            name={salesExecutive.name}
            additionalClasses={[styles.userImage]}
            imageUrl={salesExecutive.profilePhoto}
          />
        )}
      </div>
      <div className={styles.quoteWidgetWide}>
        <div className={styles.quoteHeader}>Best Rate</div>
        <h4>{quoteRate}</h4>
      </div>
    </div>
  );
};

ExtendedCustomerQuote.displayName = 'ExtendedCustomerQuote';
export default ExtendedCustomerQuote;
