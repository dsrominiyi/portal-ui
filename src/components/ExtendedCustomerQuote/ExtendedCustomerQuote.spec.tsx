import { storyRenders } from './ExtendedCustomerQuote.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('CustomerQuote', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
