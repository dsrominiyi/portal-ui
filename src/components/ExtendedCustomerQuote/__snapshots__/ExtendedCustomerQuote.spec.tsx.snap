// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerQuote component story renders match snapshots: declined 1`] = `""`;

exports[`CustomerQuote component story renders match snapshots: fullQuote 1`] = `
<div
  className="quoteContainer"
  onClick={[Function]}
>
  <div
    className="quoteWidget"
  >
    <div
      className="quoteHeader"
    >
      Reference
    </div>
    <h4>
      12345
    </h4>
  </div>
  <div
    className="quoteWidget"
  >
    <div
      className="quoteHeader"
    >
      Private Car
    </div>
    <div
      className="iconContainer"
    >
      <svg />
    </div>
  </div>
  <div
    className="quoteWidgetWide"
  >
    <div
      className="quoteHeader"
    >
      Inception Date
    </div>
    <h4>
      01/12/2019
    </h4>
  </div>
  <div
    className="quoteWidget"
  >
    <div
      className="quoteHeader"
    >
      Sales Executive
    </div>
    <UserImage
      additionalClasses={
        Array [
          "userImage",
        ]
      }
      imageUrl="/images/person-who-does-not-exist.jpg"
      name={
        Object {
          "forename": "James",
          "middleName": "Roland",
          "surname": "Fowler",
        }
      }
    />
  </div>
  <div
    className="quoteWidgetWide"
  >
    <div
      className="quoteHeader"
    >
      Best Rate
    </div>
    <h4>
      £392.63
    </h4>
  </div>
</div>
`;

exports[`CustomerQuote component story renders match snapshots: noPhoto 1`] = `
<div
  className="quoteContainer"
  onClick={[Function]}
>
  <div
    className="quoteWidget"
  >
    <div
      className="quoteHeader"
    >
      Reference
    </div>
    <h4>
      12345
    </h4>
  </div>
  <div
    className="quoteWidget"
  >
    <div
      className="quoteHeader"
    >
      Private Car
    </div>
    <div
      className="iconContainer"
    >
      <svg />
    </div>
  </div>
  <div
    className="quoteWidgetWide"
  >
    <div
      className="quoteHeader"
    >
      Inception Date
    </div>
    <h4>
      01/12/2019
    </h4>
  </div>
  <div
    className="quoteWidget"
  >
    <div
      className="quoteHeader"
    >
      Sales Executive
    </div>
    <UserImage
      additionalClasses={
        Array [
          "userImage",
        ]
      }
      name={
        Object {
          "forename": "James",
          "surname": "Fowler",
        }
      }
    />
  </div>
  <div
    className="quoteWidgetWide"
  >
    <div
      className="quoteHeader"
    >
      Best Rate
    </div>
    <h4>
      £392.63
    </h4>
  </div>
</div>
`;

exports[`CustomerQuote component story renders match snapshots: noRate 1`] = `
<div
  className="quoteContainer"
  onClick={[Function]}
>
  <div
    className="quoteWidget"
  >
    <div
      className="quoteHeader"
    >
      Reference
    </div>
    <h4>
      12345
    </h4>
  </div>
  <div
    className="quoteWidget"
  >
    <div
      className="quoteHeader"
    >
      Private Car
    </div>
    <div
      className="iconContainer"
    >
      <svg />
    </div>
  </div>
  <div
    className="quoteWidgetWide"
  >
    <div
      className="quoteHeader"
    >
      Inception Date
    </div>
    <h4>
      01/12/2019
    </h4>
  </div>
  <div
    className="quoteWidget"
  >
    <div
      className="quoteHeader"
    >
      Sales Executive
    </div>
    <UserImage
      additionalClasses={
        Array [
          "userImage",
        ]
      }
      imageUrl="/images/person-who-does-not-exist.jpg"
      name={
        Object {
          "forename": "James",
          "middleName": "Roland",
          "surname": "Fowler",
        }
      }
    />
  </div>
  <div
    className="quoteWidgetWide"
  >
    <div
      className="quoteHeader"
    >
      Best Rate
    </div>
    <h4>
      No Quote
    </h4>
  </div>
</div>
`;
